pkg/reconcilers/base_reconciler.go

```diff
@@ -0,0 +1,133 @@
+package reconcilers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = logf.Log.WithName("reconcilers")
+
+// MutateFn is a function which mutates the existing object into it's desired state.
+type MutateFn func(existing, desired common.KubernetesObject) (bool, error)
+
+func CreateOnlyMutator(existing, desired common.KubernetesObject) (bool, error) {
+	return false, nil
+}
+
+type BaseReconciler struct {
+	mgr             manager.Manager
```

Comment:
We used to pass the client and scheme directly. Is it a good idea to pass the whole manager dependency?
When doing tests how are we going to mock this? When we passed the client we could use the FakeClient and pass it.

Reply:
I was thinking about mocking Manager with fake client. 

But, I agree. No big benefit. Will add client and scheme as fields.